superdesk/commands/flush_elastic_index.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+import requests
+from urllib.parse import urljoin
+from flask import current_app as app
+import superdesk
+from content_api import ELASTIC_PREFIX as CAPI_ELASTIC_PREFIX
+
+from .index_from_mongo import IndexFromMongo
+
+# this one is not configurable
+SD_ELASTIC_PREFIX = 'ELASTICSEARCH'
+
+
+class FlushElasticIndex(superdesk.Command):
+    """Flush elastic index.
+
+    It removes elastic index, creates a new one and index it from mongo.
+    You must specify at least one elastic index to flush:
+    ``--sd`` (superdesk) or ``--capi`` (content api)
+    """
+
+    option_list = [
+        superdesk.Option('--sd', action='store_true', dest='sd_index'),
+        superdesk.Option('--capi', action='store_true', dest='capi_index')
+    ]
+
+    def run(self, sd_index, capi_index):
+        if not (sd_index or capi_index):
+            raise SystemExit('You must specify at least one elastic index to flush. '
+                             'Options: `--sd`, `--capi`')
+        if sd_index:
+            self._delete_elastic(superdesk.app.config['ELASTICSEARCH_INDEX'])
+        if capi_index:
+            self._delete_elastic(superdesk.app.config['CONTENTAPI_ELASTICSEARCH_INDEX'])
+
+        self._index_from_mongo(sd_index, capi_index)
+
+    def _delete_elastic(self, index):
+        """Deletes elastic index
+
+        :param str index: elastix index
+        :raise: SystemExit exception if delete elastic index response status is not 200 or 404.
+        """
+        es_index_url = urljoin(
+            superdesk.app.config['ELASTICSEARCH_URL'],
+            index
+        )
+        print('- Removing elastic index "{}"'.format(index))
+        resp = requests.delete(es_index_url)
```

Comment:
you can get [elasticsearch instance](https://github.com/superdesk/superdesk-core/blob/master/superdesk/commands/rebuild_elastic_index.py#L41) and then you can use [indices api](https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.client.IndicesClient.delete) so you wouldn't have to deal with different errors here, it would raise an exception like index not found

Reply:
@petrjasek thanks, I will use it.

Reply:
@petrjasek , I used elasticsearch instance to operate with search. In this case it is kinda the same, because I want to print log when index was not found, so I still need to deal with exceptions :) But it is more solid to use a wrapper.